File: lib/contact_events.dart
abstract class ContactEvent {}

class ContactsChangedEvent implements ContactEvent {
  @override
  String toString() {
    return 'ContactsChangedEvent{}';
  }
}

class UnknownContactEvent implements ContactEvent {
  final Map<String, dynamic> payload;

  UnknownContactEvent(this.payload);

  @override
  String toString() {
    return 'UnknownContactEvent{payload: $payload}';
  }
}

File: android/src/main/kotlin/co/sunnyapp/flutter_contact/ContactEvents.kt
package co.sunnyapp.flutter_contact

sealed class ContactEvent {
    abstract fun toMap(): Struct
}

data class ContactChangedEvent(val contactId: String) : ContactEvent() {
    override fun toMap(): Struct {
        return mapOf("event" to "contact-changed",
                "contactId" to contactId)
    }
}

object ContactsChangedEvent : ContactEvent() {
    override fun toMap(): Struct {
        return mapOf("event" to "contacts-changed")
    }
}